Upgraded xtext version to 2.9 in infomodel. Created web app for infor…

shiv12095 opened the pull request:
…mation model text editor

Signed-off-by: shiv12095 <shiv12095@iiitd.ac.in>

aedelmann commented:
Hi Shiv, 
does this PR solve the formatter issue also?

shiv12095 commented:
No this doesn't. Will fix the issue and update the PR

aedelmann commented:
Hi Shiv, 
each pom file of your added submodule states the xtext version 2.9.0 but the parent pom actually defines a xtext version property 2.7.1 that all sub maven modules inherit. In your case it could work the same way if you just change the parent property.
cheers, alex

shiv12095 commented:
I  this PR I changed the xtext version in the parent pom file and removed the xtext version property from the submodules. The repository builds fine.  Still need to work on the Formatter tests

aedelmann commented:
Is there a reason why you are referencing xtext 2.9.0 instead of the latest 2.9.2 ?

shiv12095 commented:
Fixed the formatter tests. 
@aedelmann There was no specific reason for 2.9 . Should I migrate it to 2.9.2 ?

aedelmann commented:
Yep. Latest is greatest :)

aedelmann commented on bundles/org.eclipse.vorto.editor.infomodel.web/pom.xml:
```diff
@@ -0,0 +1,146 @@
+		<dependency>
+			<groupId>org.webjars</groupId>
+			<artifactId>requirejs</artifactId>
+			<version>2.1.20</version>
```
Can you please use requirejs 2.1.15 instead since there is already a CQ and i can piggyback that. 
Thanks!

aedelmann commented on bundles/org.eclipse.vorto.editor.infomodel.web/pom.xml:
```diff
@@ -0,0 +1,146 @@
+			<version>2.1.20</version>
+		</dependency>
+		<dependency>
+			<groupId>org.webjars</groupId>
```
Can you please use jquery 2.1.1 since we already have a CQ approved for that :) Thanks.

aedelmann commented on bundles/org.eclipse.vorto.editor.infomodel.web/pom.xml:
```diff
@@ -0,0 +1,146 @@
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-simple</artifactId>
+			<version>1.7.12</version>
```
Can you use slf4j-api 1.7.13 since we have a CQ approved for that already. Thanks!

aedelmann commented on bundles/org.eclipse.vorto.editor.infomodel.web/src/org/eclipse/vorto/editor/infomodel/web/ServerLauncher.xtend:
```diff
@@ -0,0 +1,68 @@
+ * This program starts an HTTP server for testing the web integration of your DSL.
+ * Just execute it and point a web browser to http://localhost:8080/
+ */
+class ServerLauncher {
```
Can you remove this class as it is not needed in the future due to usage of springboot. Can you also verify and clean unneeded dependencies in the pom.xml ?
Thanks!

shiv12095 commented:
I'll update the files according to the CQ and push the changes

aedelmann commented on bundles/org.eclipse.vorto.editor.infomodel.web/WebRoot/index.html:
```diff
@@ -0,0 +1,44 @@
+<html>
+<head>
+	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
```
Can this file also be deleted as it is not required?

aedelmann commented on bundles/org.eclipse.vorto.editor.infomodel.web/pom.xml:
```diff
@@ -0,0 +1,138 @@
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.slf4j</groupId>
```
Can this be changed to sl4j-api and please use the same library version as the current vorto repository is using.

aedelmann commented:
Hi Shiv, 
I added some additional comments in your commits. One more thing: Can you squash your 30 commits into a single commit, sign off that one and Pull Request that?
Thank you!

Cheers, Alex

shiv12095 commented:
I created a new pull request. All the changes have been made in a single commit now. The pull request is here at : https://github.com/eclipse/vorto/pull/227

I will delete this request later
